docs/en/sampling_params.md

```diff
@@ -142,3 +165,252 @@ print(response.json())
 
 The `image_data` can be a file name, a URL, or a base64 encoded string. See also `python/sglang/srt/utils.py:load_image`.
 Streaming is supported in a similar manner as [above](#streaming).
+
+## Performance Implications on Penalties
+
+While you can apply penalties by supplying relevant `sampling_params`, this comes with some drawbacks.
+
+These drawbacks will be applied to every single requests in the same batch, as penalizers also applies in batch.
+
+### Latency
+
+While we try to compute penalty algorithms through CUDA, it is still additional computation on top of the basic sampling logic. For detailed overhead, we recommend you to run your own benchmarks, but you can find samples below to get a glimpse.
+
+### Memory
+
+Since we compute penalty algorithms through CUDA, the logic stores relevant parameters on GPU. This is usually in a scale of `vocab_size` multiplied by `running_requests`.
+
+You can run your own benchmark with desired parameters on your own hardware to make sure it's not OOMing before using.
+
+Tuning `--mem-fraction-static` and/or `--max-running-requests` will help. See [here](hyperparameter_tuning.md#minor-tune---max-prefill-tokens---mem-fraction-static---max-running-requests) for more information.
+
+### Benchmarks
```

Comment:
I wouldn't strongly recommend including the benchmark results in the document, but if necessary, consider using a collapsible section so users can expand it when needed.

```
<details>
<summary>title</summary>
</details>
```

Reply:
It is already wrapped with collapsible section!